packages/plugin-core/src/WSUtilsV2.ts

```diff
@@ -95,6 +96,48 @@ export class WSUtilsV2 implements IWSUtilsV2 {
     });
   }
 
+  /**
+   * See {@link IWSUtilsV2.findNoteFromMultiVaultAsync}.
+   */
+  async findNoteFromMultiVaultAsync(opts: { fname: string; vault?: DVault }) {
+    const { fname, vault } = opts;
+    let existingNote: NoteProps | undefined;
+    const engine = ExtensionProvider.getEngine();
+    const maybeNotes = NoteUtils.getNotesByFnameFromEngine({
+      fname,
+      engine,
+      vault,
+    });
+
+    if (maybeNotes.length === 1) {
+      // Only one match so use that as note
+      existingNote = maybeNotes[0];
+    } else if (maybeNotes.length > 1) {
+      // If there are multiple notes with this fname, prompt user to select which vault
+      const vaults = maybeNotes.map((noteProps) => {
+        return {
+          vault: noteProps.vault,
+          label: VaultUtils.getName(noteProps.vault),
+        };
+      });
+
+      const items = vaults.map((vaultPickerItem) => ({
+        ...vaultPickerItem,
+        label: vaultPickerItem.label
```

Comment:
There might be a small bug here:

![Screenshot_20220128_150011](https://user-images.githubusercontent.com/1008124/151613567-d756b846-9e2e-4042-ba3a-e2ab70ec144a.png)

Steps to reproduce:
- Open the test workspace. You may need to close and restart after opening it for the first time.
- Use the lookup bar, type "templates.book", select the "prompt for vault" button (box shape, on the very left of the top bar), then select "Create New" and hit enter.
- Write something in the new template note.
- Create a note "book.something", the prompt will have the wrong vault names.

The problem goes away after restarting though. I would almost suspect `getNotesByFnameFromEngine` returning an incorrect result, but it does return 2 notes hence the 2 vaults.

Reply:
Also a nitpick, but can we make it clearer what's being selected here? Maybe this function should accept a parameter to override the item labels and title or something, but this is especially confusing if you enable the prompt for which vault the note should be created in.

Something like "Select which template to apply" as the title, and the labels being something like "templates.book from vault1" would be easier to understand I think. Feel free to leave that to a future task though, I don't want to block you from merging this feature.

Reply:
I like that suggestion. I'll go ahead and update the title/labels

Reply:
Regarding the incorrect vault name, it looks like an indexing issue with the engine. For example, after I create a new note in vaultThree, when I inspect `engine.notes`, it thinks the vault is vault1. Restarting fixes the vault name